src/yoroi-wallets/storage.ts

```diff
@@ -44,3 +44,111 @@ export const makeStorageWithPrefix = (prefix: string): Storage => {
     },
   }
 }
+
+interface PrefixedStorage extends Omit<AsyncStorageStatic, 'multiMerge' | 'mergeItem'> {
+  path: string
+  toAbsolutePath: (key: string) => string
+}
+
+export const makeRootStorage = (storage: AsyncStorageStatic = AsyncStorage) => {
+  return {
+    ...storage,
+    path: '/',
+    toAbsolutePath: (key: string) => `/${key}`,
+  }
+}
+
+export const debug = async () => {
+  const keys = await AsyncStorage.getAllKeys()
+  const items = await AsyncStorage.multiGet(keys)
+
+  const data = Object.fromEntries(items)
+
+  return data
+}
+
+export const createPrefixedStorage = (options?: {storage?: PrefixedStorage; path?: string}) => {
+  const {storage = makeRootStorage(), path = '/'} = options || {}
+  if (!validatePath(path)) throw new Error("invalid path: paths may not contain '/'")
+  const isRootStorage = storage.path === '/'
+  const isRootPath = path === '/'
+
+  const storagePath =
+    isRootStorage && isRootPath
+      ? '/'
+      : !isRootStorage && isRootPath
+      ? storage.path
+      : isRootStorage && !isRootPath
+      ? `${storage.path}${path}`
+      : `${storage.path}/${path}`
+
+  const prefixedStorage: PrefixedStorage = {
+    path: storagePath,
+    toAbsolutePath: (fileName: string) => toAbsolutePath(storage.path, path, fileName),
+    async getItem(fileName: string) {
+      const absolutePath = toAbsolutePath(storage.path, path, fileName)
+      return AsyncStorage.getItem(absolutePath)
+    },
+    async multiGet(fileNames: string[]) {
+      const absolutePaths = fileNames.map((fileName) => toAbsolutePath(storage.path, path, fileName))
+      const items = await AsyncStorage.multiGet(absolutePaths)
+      return items.map(([absolutePath, file]) => [toFileName(absolutePath), file])
+    },
+
+    async setItem(fileName: string, file: string) {
+      const absolutePath = toAbsolutePath(storage.path, path, fileName)
+      return AsyncStorage.setItem(absolutePath, file)
+    },
```

Comment:
It looks like when `fileName` is something like `example/folder/test.jpg`, then `toFileName` will return just `test.jpg`. Should we add a check to `fileName` to ensure it does not contain `/`?